_posts/2022-06-14-infinispan-14-indexing-query-news.adoc

```diff
@@ -70,24 +70,23 @@ public class Author {
 }
 ----
 
-Indexed fields without any special string/text transformation will be annotated as `@Basic`.
-If we need to apply a normalizer to a String field, we will opt for a `@Keyword` annotation.
-If we need to apply an analyzer to a String field, we will opt for a `@Text` annotation.
+Use the `@Basic` annotation for indexed fields without any special string/text transformation.
+When you apply a normalizer to a String field, use the `@Keyword` annotation.
+When you apply an analyzer to a String field, use the `@Text` annotation.
 
-The new annotations allow setting with the same annotation if the field should be sortable, or projectable, or its normalizer or its analyzer.
-However, not all the combinations will be possible, for instance the attribute sortable is not present on the `@Text` annotation,
-since an analyzed field cannot be used to sort the result set.
+You can use the same annotations for the fields that are sortable or projectable and for its normalizer or analyzer.
+However, not all the combinations are possible, for instance the attribute `sortable` is not present on the `@Text` annotation,
+since an analyzed field cannot sort the result set.
 
-`indexNullAs` attribute allow now to define a default value to use on index in case the corresponding entity values was null.
+When indexing entities with null values, define a default value with the `indexNullAs` attribute.
 
-Embedded indexes are defined using the `@Embedded` annotation, and it is possible to choose between the `NESTED` structure which
-preserves the original object relationship structure and `FLATTENED` structure which makes the leaf fields multi-valued
-fields of the parent entity.
+For embedded indexes use the `@Embedded` annotation and choose between two structures.
+The `NESTED` structure preserves the original object relationship structure and the `FLATTENED` structure makes the multi-valued fields of the parent instance into leaf fields.
 
 == Index startup mode
 
-Sometimes indexes can be persistent and cache data not or vice versa, for those cases it can be useful
-to perform some operations to ensure the index will be consistent with data in the cache.
+Indexes can be persistent or volatile just like the stored cache data.
```

Comment:
I think that we have here both the possibilities.
* indexes are persisted and cache data is volatile
* indexes are volatile and cache data is persisted.

Reply:
Indexes can be persistent and cache data can be volatile and vice versa. You can perform some operations to keep the index consistent with data in the cache. Possible values are `purge`, `reindex`, `none`,and `auto`. When you choose `auto` mode, Infinispan decides what is the right operation to run to align cache data and indexes.